lib/racecar/consumer.rb

```diff
@@ -34,13 +34,22 @@ def subscribes_to(*topics, start_from_beginning: true, max_bytes_per_partition:
 
     def configure(producer:, consumer:, instrumenter: NullInstrumenter)
       @producer = producer
+      @delivery_handles = []
+
       @consumer = consumer
       @instrumenter = instrumenter
     end
 
     def teardown; end
 
-    # Delivers messages that got produced.
+    # Blocks until all messages produced so far have been successfully published. There
+    # are two scenarios where the function raises:
+    # Rdkafka::AbstractHandle::WaitTimeoutError: reached the default timeout of 60s. You
+    #     can call deliver! again to wait longer.
+    # Rdkafka::RdkafkaError: delivery failed for the reason in the exception. Some of
+    #     them include permanent failures, e.g. "msg_timed_out" which librdkafka sets
+    #     after failing to deliver the message within "message.timeout.ms". The caller
+    #     must handle these edge cases or run into head of line blocking.
```

Comment:
Would you be up for mapping the Rdkafka exceptions to Racecar native exceptions with good error messages and UX?